app/workers/notifications/notifiable_action_worker.rb

```diff
@@ -0,0 +1,14 @@
+module Notifications
+  class NotifiableActionWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :low_priority, retry: 10
+    def perform(notifiable_id, notifiable_type, action)
+      # checking type, but leaving space for notifyable types
+      return unless notifiable_type == "Article"
+
+      notifiable = notifiable_type.constantize.find_by(id: notifiable_id)
```

Comment:
@mstruve @rhymes This has nothing to do with @serenaf's PR per se, but I'd like to address this pattern of using `find_by` and doing nothing if `nil` is returned. Given that we're looking up by ids, this shouldn't happen all that often, generally only if the object got removed before the job ran. So I think it might be a good idea to have some insights into this behavior, something like

```ruby
def perform(notifiable_id, notifiable_type, action)
  return unless notifiable_type == "Article"
  notifiable = notifiable_type.constantize.find(notifiable_id)
  Notifications::NotifiableAction::Send.call(notifiable, action)
rescue ActiveRecord::RecordNotFound
  # notify an exception tracker, log to statsd or something
end
```

If we don't want to use exceptions (though in this case it keeps the happy path free of conditionals) we can also do something like:

```ruby
unless notifiable
  # notify an exception tracker, log to statsd or something
  return
end

Notifications::NotifiableAction::Send.call(notifiable, action)
```

Either way, I think it would be good to have a little more insight into things that should not happen™.

Reply:
I think the plan is still to review all workers when transitioned to see which ones should have their errors "propagated" but to keep them as is in the meantime.

Sidekiq also has some error handling tricks that can be employed: https://github.com/mperham/sidekiq/wiki/Error-Handling

Reply:
@citizen428 yeah, I personally hate the silent failing but we decided that maintaining the status quo was probably best for the initial move and then you bet we will be changing up how we alert and handle certain errors.

Reply:
Great, thanks for the context @rhymes and @mstruve. Happy to be part of any discussion on how we deal with this in the future.